### ISSUE TITLE
linter: no-unused-vars is incorrectly reported within a rest spread

### ISSUE THREAD

**Issue by posva** [oxc-project/oxc#11640]
### What version of Oxlint are you using?

1.1.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` config file look like?

```jsonc
// noconfig
```

### What happened?

It fails with this code:

```ts
 const { a, b, ...rest } = { a: 0, b: 2, c: 5, d: 7 }
 console.log({ a, rest }) 
```

`b` here is used to remove it from `rest`

**Comment by camc314** [comment]
Hi, thanks for using oxlint, to allow rest siblings, please configure your `.oxlintrc.json` configuration like so

```
{
    "rules": {
        "no-unused-vars": [
            "error",
            {
                "ignoreRestSiblings": true
            }
        ]
    }
}

```


https://github.com/oxc-project/oxc/blob/cc0112f2ad57df2fe15f4a043fcd1d9bad6558e4/crates/oxc_linter/src/rules/eslint/no_unused_vars/options.rs#L70-L84

**Comment by posva** [comment]
I think it should be the default because it’s not technically unused, it’s used to be removed from rest, like writing `delete rest.other`

**Comment by camc314** [comment]
Thanks for the feedback.

I think for now, I am going to maintain the same behaviour as  typescript-eslint untill we recieve more feedback. As another alternative, you could also write code like:
```
var { foo: _foo, ...objWithoutFoo } = obj
```

Writing it like this, and having `^_*`  makes it arguably clearer that that `obj` is being destructured with the `foo`  property removed

**Comment by posva** [comment]
I understand you might want to follow typescript-eslint strictly, but this feels like a missed opportunity for a better default that should not create new listing errors for projects adopting oxc. For example, someone starts using oxc without a config, the default to `true` seems to simply make more sense.

BTW it's even in [the proposed configuration](https://typescript-eslint.io/rules/no-unused-vars/#what-benefits-does-this-rule-have-over-typescript) in their docs.
